### PR TITLE
[master] MGMT-20756: assisted installer naive string concatenation for partition finding

### PR DESCRIPTION
**Pull request by shay23bra**
## Problem
Assisted installer used naive string concatenation for partition finding, causing failures with device mapper devices (dm-*). For example, trying to mount `/dev/dm-02` instead of the correct `/dev/dm-2`.

## Solution  
Replace string concatenation with robust `lsblk --json` based partition discovery that works with all device types:
- Device mapper: `/dev/dm-0` → `/dev/dm-2`, `/dev/dm-3`, `/dev/dm-4`
- Standard SATA: `/dev/sda` → `/dev/sda2`, `/dev/sda3`, `/dev/sda4`  
- NVMe: `/dev/nvme0n1` → `/dev/nvme0n1p2`, `/dev/nvme0n1p3`, `/dev/nvme0n1p4`
- MMC: `/dev/mmcblk1` → `/dev/mmcblk1P2`, `/dev/mmcblk1P3`, `/dev/mmcblk1P4`

## Changes
- Add `getPartitionPathFromLsblk()` function using `lsblk --json` output parsing
- Update `OverwriteOsImage()` to use new partition discovery method
- Fix `calculateFreePercent()` to use proper array indexing  
- Add comprehensive unit tests (16 tests) covering all device types and edge cases
- Add integration test for device mapper devices

## Testing
- ✅ All existing tests pass (maintains backward compatibility)
- ✅ New device mapper test passes 
- ✅ Comprehensive unit test coverage for new functionality
- ✅ Verified exact ticket scenario: `/dev/dm-0` partition 2 correctly returns `/dev/dm-2`

Fixes: [MGMT-20756](https://issues.redhat.com//browse/MGMT-20756)

### CODE REVIEW

File: src/ops/ops.go [hunk 1]
```diff
@@ -1033,6 +1033,52 @@ func stripDev(device string) string {
...
+	type node struct {
+		Name     string  `json:"name"`
+		Size     int64   `json:"size"`
+		Children []*node `json:"children"`
+	}
+	var disks struct {
+		Blockdevices []*node `json:"blockdevices"`
+	}
+
+	ret, err := o.ExecPrivilegeCommand(nil, "lsblk", "-b", "-J")
```
Comment (omertuc) [inline src/ops/ops.go:1049]: Avoid short flags in code

```suggestion
	ret, err := o.ExecPrivilegeCommand(nil, "lsblk", "--bytes, "--json")
```

Comment (omertuc) [reply src/ops/ops.go:1049]: Also we can pass lsblk the device and it will give us only the information about that device

```suggestion
	ret, err := o.ExecPrivilegeCommand(nil, "lsblk", "-b", "-J", device)
```

**Review by omertuc — CHANGES_REQUESTED** [review]

**Review by omertuc — APPROVED** [review]
